constraint/pkg/client/client.go

```diff
@@ -431,14 +533,20 @@ func (c *Client) AddData(ctx context.Context, data interface{}) (*types.Response
 			continue
 		}
 
-		err = c.driver.AddData(ctx, name, key, processedDataCpy)
-		if err != nil {
-			errMap[name] = err
+		// To avoid maintaining duplicate caches, only Rego should get its own
+		// storage. We should work to remove the need for this special case
+		// by building a global storage object. Right now Rego needs its own
+		// cache to cache constraints.
+		if _, ok := c.drivers[regoSchema.Name]; ok {
```

Comment:
what happens if !ok is returned?

Reply:
This code is just saying "only call add/remove data for the Rego driver".

Having multiple drivers maintain separate caches would lead to excess RAM usage.

Ultimately, what we need is a cache that can be shared across drivers, but that is quite a bit of work, and not necessary for the CEL KEP (which has no notion of referential constraints).

Reply:
should we at least log it if it's not a rego driver?

Reply:
My concern there would be log verbosity. For example: someone who uses Gatekeeper who watches pods would see a log line every time a pod was updated.

Reply:
OTOH, Gatekeeper would have the rego driver, so no need to log

Reply:
https://github.com/open-policy-agent/frameworks/pull/293/commits/19851c7e118f61a744af77636fed70807001cf4d LGTM. Can you pls resolve the file conflict?

Reply:
Thanks! Resolved.